web/src/main/ui/app/components/flows-new/edit-flow/mapping/ui/mapping-ui.component.ts

```diff
@@ -145,58 +145,58 @@ export class MappingUiComponent implements OnChanges {
   getMapValidationErrors(sourceURI?: string) {
     let self = this;
     self.isTestClicked = true;
-    self.manageFlowsService.getMap(self.mapping.name).subscribe((map: any) => {
-
-      self.manageFlowsService.getMappingErrors(map.name, map).subscribe(resp => {
-
-        self.mapErrors = resp;
-        if (JSON.stringify(self.mapErrors) != JSON.stringify({})) {
-          //checking if the response contains any error
-          self.checkKeyinObject(self.mapErrors, "errorMessage");
-
-          if (!self.containErrors) {
-            let uri;
-            if (sourceURI) {
-              uri = sourceURI;
-            }
-            else {
-              uri = map.sourceURI;
-            }
-            this.manageFlowsService.testMap(map.name, String(map.version), uri).subscribe(resp => {
-              console.log("testMap called");
-              this.mapResults = resp;
-              delete this.mapResults["info"];
-              this.mapResults = this.mapResults[this.entityName];
-              console.log(this.mapResults);
-            },
-              err => {
-                if (err.hasOwnProperty('error')) {
-                  console.log("found error");
-                  if (err['error']['code'] == 500) {
-                    let result = self.dialogService.alert(
-                      'Please try again in a few seconds. Thanks for your patience.',
-                      'OK'
-                    );
-                    result.subscribe();
+    setTimeout(function(){ 
+      self.manageFlowsService.getMap(self.mapping.name).subscribe((map: any) => {
+        self.manageFlowsService.getMappingErrors(map.name, map).subscribe(resp => {
+          self.mapErrors = resp;
+          if (JSON.stringify(self.mapErrors) != JSON.stringify({})) {
+            //checking if the response contains any error
+            self.checkKeyinObject(self.mapErrors, "errorMessage");
+  
+            if (!self.containErrors) {
+              let uri;
+              if (sourceURI) {
+                uri = sourceURI;
+              }
+              else {
+                uri = map.sourceURI;
+              }
+              self.manageFlowsService.testMap(map.name, String(map.version), uri).subscribe(resp => {
+                console.log("testMap called");
+                self.mapResults = resp;
+                delete self.mapResults["info"];
+                self.mapResults = self.mapResults[self.entityName];
+                console.log(self.mapResults);
+              },
+                err => {
+                  if (err.hasOwnProperty('error') && String(err['error']['message']).indexOf("Could not find mapping") >= 0) {
+                    console.log("found error");
+                    if (err['error']['code'] == 500) {
+                      let result = self.dialogService.alert(
+                        'Please try again in a few seconds. Thanks for your patience.',
```

Comment:
I recommend a more specific message here so the user isn't left wondering what's going on - perhaps:

"QuickStart hasn't finished loading the updated mapping. Please try again."

Though if we detect this error, couldn't QS wait a few times itself - or is that what the 750ms delay below is for?

Reply:
@rjrudin ,
1. Whenever someone hits "Test" button, I add a delay for 750 ms (regardless of whether the mapping has changed or not, the assumption is more often than not the user will hit the button after changing the mapping) and then make the request. I don't retry in case if mapping is not found. Is that logic ok ?
2. I will reword the message in the pop up